iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -242,9 +242,7 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     //     initHandle(&handle);
     //   }
     bool m_hasValue{false};
-    // safe access is guaranteed since the array is wrapped inside the optional
-    // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
-    alignas(T) byte_t m_data[sizeof(T)];
+    containers::UninitializedArray<T, 1> m_data;
```

Comment:
@MatthiasKillat @FerdinandSpitzschnueffler what do you think about `UninitializedCell<T>`? This would take care of the alignment and the type-safety? The `UninitializedArray` would then be just a thin wrapper around a `UninitializedCell<T> m_data[N]`. I might be more expensive when we want to use it with something like a `char` tough, e.g. `UninitializedCell<char> m_data[n]`. One would need to check if this is really the case

Reply:
@elBoberido The difference from `std::aligned_storage` would be that it is typed. Not sure it is worth the trouble just for that. Note that we already use `aligned_storage`, so there are precedent. The `UnitializedCell<T>` would need to be tested (even if trivial).

Reply:
I decided to replace the `UninitializedArray` with the `aligned_storage`, mainly because I think having an `UninitializedCell` and the `UninitializedArray` as wrapper is at the moment overkill.